opentelemetry-sdk/src/resource/env.rs

```diff
@@ -76,15 +76,15 @@ pub struct SdkProvidedResourceDetector;
 impl ResourceDetector for SdkProvidedResourceDetector {
```

Comment:
not something added in this PR, this `SdkProvidedResourceDetector` does not sound very clear. Maybe we should remove that pub struct, and provide a method like `add_default_resource` instead.
We also need to have a test for each providers(log/trace/meter) to ensure that
1. If users specifies *no* explicit resource (neither in code, nor in env vars), then the default is added
2. If users specifies *no* explicit resource in code, but env vars, then we get the env vars one.
3. Users can have resource without the default.
4. Users can have defaults + their own.
5. Validate precedence in all scenarios.
+ anything else missing.

This is a good candidate for follow up PR @rogercoll if you are interested in continuing this work!

Reply:
We already have it coverage for `TracerProvider` [here](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/trace/provider.rs#L277)

Reply:
Hmm I don't think we need explicit need a method to add default resource. It sounds like `Default` trait is a better fit and we are using it.

Reply:
> We already have unit tests mentioned above for `TracerProvider` [here](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/trace/provider.rs#L277)

Thanks for pointing. We need it for all signals. Also https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/trace/provider.rs#L291 is only validating if service name is populated, it should do telemetry resource as well, right?

Reply:
Ah good point. Yeah we probably need to expand the coverage to other signal and other resources

Reply:
Make sense to me, I will follow up on this topic for the other signals.